x/name/client/cli/gov.go

```diff
@@ -106,7 +110,93 @@ $ %s tx gov submit-proposal param-change tx gov submit-proposal \
 		},
 	}
 
-	cmd.Flags().String(FlagOwner, "", "The owner of the new name, optional (defaults to from address")
+	cmd.Flags().String(FlagOwner, "", "The owner of the new name, optional (defaults to from address)")
+	cmd.Flags().BoolP(flagRestricted, "r", true, "Restrict creation of child names to owner only, optional (default false)")
```

Comment:
This `flagRestricted` needs some love both here and in the new `GetModifyNameProposalCmd ` function.

If you run `provenanced tx gov submit-legacy-proposal root-name-proposal --help`, you'll see this in the flags listing:
```
  -r, --restrict                 Restrict creation of child names to owner only, optional (default false) (default true)
```

1. The "(default false)" in the description is just flat out incorrect. And boolean flags should always be optional, so the whole `", optional (default false)"` part should be removed.
2. Cobra automatically adds a "(default <val>)" when it's something of interest (e.g not `""`).
3. Boolean flags should default to `false`. The absence of the flag as an arg should be the default behavior. The presence of a flag as an arg should change that behavior. When you default a boolean flag to `true`, that gets messed up; absence of the flag is now the same as providing it. In order to override the default, it would have to be provided as `--restricted=false` (you can't even do `--restricted false`).

TL;DR (too late!):
This should either default to `false`, or, if the same current behavior is desired, should be renamed to `--unrestricted` so that it can have a `false` default value. Either way the description should be cleaned up too.

See also: The `--iavl-disable-fastnode` debacle.

Reply:
I cleaned up the help text for this. Also, renamed the flag to `--unrestrict` in the BindName request, RootName proposal, and in this new proposal. I also made sure to mention this in the README as well.